app/src/main/java/org/example/viotester/AlgorithmWorker.java

```diff
@@ -103,8 +110,28 @@ void start() {
         }
 
         void stop() {
-            if (mSettings.recordGps || mSettings.recordWiFiLocations)
+            if (mSettings.recordGps || mSettings.recordWiFiLocations) {
+                locationManager.unregisterGnssMeasurementsCallback(this);
```

Comment:
Logically, should only happen on `mSettings.recordGps` but not if only `mSettings.recordWiFiLocations`, I think